extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/sql/ThetaSketchSetBaseOperatorConversion.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.datasketches.theta.SketchSetPostAggregator;
+import org.apache.druid.sql.calcite.expression.DruidExpression;
+import org.apache.druid.sql.calcite.expression.OperatorConversions;
+import org.apache.druid.sql.calcite.expression.PostAggregatorVisitor;
+import org.apache.druid.sql.calcite.expression.SqlOperatorConversion;
+import org.apache.druid.sql.calcite.planner.Calcites;
+import org.apache.druid.sql.calcite.planner.PlannerContext;
+import org.apache.druid.sql.calcite.table.RowSignature;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class ThetaSketchSetBaseOperatorConversion implements SqlOperatorConversion
+{
+  public ThetaSketchSetBaseOperatorConversion()
+  {
+  }
+
+  @Override
+  public SqlOperator calciteOperator()
+  {
+    return makeSqlFunction();
+  }
+
+  @Nullable
+  @Override
+  public DruidExpression toDruidExpression(
+      PlannerContext plannerContext,
+      RowSignature rowSignature,
+      RexNode rexNode
+  )
+  {
+    return null;
+  }
+
+  @Nullable
+  @Override
+  public PostAggregator toPostAggregator(
+      PlannerContext plannerContext,
+      RowSignature rowSignature,
+      RexNode rexNode,
+      PostAggregatorVisitor postAggregatorVisitor
+  )
+  {
+    final List<RexNode> operands = ((RexCall) rexNode).getOperands();
+    final List<PostAggregator> inputPostAggs = new ArrayList<>();
+    Integer size = null;
+
+    int operandCounter = 0;
+    for (RexNode operand : operands) {
+      final PostAggregator convertedPostAgg = OperatorConversions.toPostAggregator(
+          plannerContext,
+          rowSignature,
+          operand,
+          postAggregatorVisitor
+      );
+      if (convertedPostAgg == null) {
+        if (operandCounter == 0) {
+          try {
+            size = RexLiteral.intValue(operand);
+          }
+          catch (RuntimeException re) {
+            return null;
+          }
+        } else {
+          return null;
+        }
+      } else {
+        inputPostAggs.add(convertedPostAgg);
+        operandCounter++;
+      }
+    }
+
+    return new SketchSetPostAggregator(
+        postAggregatorVisitor.getOutputNamePrefix() + postAggregatorVisitor.getAndIncrementCounter(),
+        getSetOperationName(),
+        size,
+        inputPostAggs
+    );
+  }
+
+  private SqlFunction makeSqlFunction()
+  {
+    return new SqlFunction(
+        getFunctionName(),
+        SqlKind.OTHER_FUNCTION,
+        ReturnTypes.explicit(
+            factory -> Calcites.createSqlType(factory, SqlTypeName.OTHER)
+        ),
+        null,
+        OperandTypes.VARIADIC,
+        SqlFunctionCategory.USER_DEFINED_FUNCTION
+    );
+  }
+
+  public String getSetOperationName()
+  {
+    throw new UnsupportedOperationException("getSetOperationName() is not implemented.");
```

Comment:
Suggest `public abstract String getSetOperationName()`.

Reply:
Changed this to abstract method